src/include/duckdb.h

```diff
@@ -5960,6 +5986,83 @@ The result must be destroyed with `duckdb_destroy_logical_type`.
 DUCKDB_C_API duckdb_logical_type duckdb_table_function_bind_get_result_column_type(duckdb_bind_info info,
                                                                                    idx_t col_idx);
 
+//----------------------------------------------------------------------------------------------------------------------
+// Catalog Interface
+//----------------------------------------------------------------------------------------------------------------------
+// DESCRIPTION:
+// Functions to access catalog entries
```

Comment:
Can we elaborate a bit here on how to use this interface / why you would use it.